app/services/store.js

```diff
@@ -19,11 +19,19 @@ export default class ExtendedStoreService extends Store {
 
   async queryAll(modelName, query, options) {
     query = query || {}; // eslint-disable-line no-param-reassign
-    const count = await this.count(modelName, query, options);
     const batchSize = query.page?.size || 100;
+
+    const firstBatch = this.query(modelName, Object.assign({}, query, {
+      'page[size]': batchSize,
+      'page[number]': 0,
+    }));
+
```

Comment:
**nitpick:** Never used `Object.assign` so I might be wrong but I think this can be rewritten as
```
this.query(modelName, {
    ...query
    'page[size]': batchSize,
    'page[number]': 0,
}
```